conans/client/graph/proxy.py

```diff
@@ -24,6 +23,14 @@ def __init__(self, client_cache, output, remote_manager):
         self._registry = client_cache.registry
 
     def get_recipe(self, conan_reference, check_updates, update, remote_name, recorder):
+        if self._client_cache.installed_as_editable(conan_reference):
+            conanfile_path = self._client_cache.conanfile(conan_reference)
+            status = RECIPE_EDITABLE
+            # TODO: log_recipe_got_from_editable(reference)
+            # TODO: recorder.recipe_fetched_as_editable(reference)
+            metadata = self._client_cache.load_metadata(conan_reference)
```

Comment:
I don't think an editable should have revision at all, specially not the revision of the package in the cache, that almost by definition is different than the edited one.

Reply:
(Comment correspond to the next line in the source)

If I return `conan_reference` or `conan_reference.copy_clear_rev()`, then I get errors in some tests (`functional/editable/commands/info_on_child_test`):

```
Traceback (most recent call last):
  File "/Users/jgsogo/dev/conan/conan/conans/client/command.py", line 1481, in run
    method(args[0][1:])
  File "/Users/jgsogo/dev/conan/conan/conans/client/command.py", line 578, in info
    self._outputer.info(deps_graph, only, args.package_filter, args.paths)
  File "/Users/jgsogo/dev/conan/conan/conans/client/conan_command_output.py", line 89, in info
    node_times=self._read_dates(deps_graph),
  File "/Users/jgsogo/dev/conan/conan/conans/client/conan_command_output.py", line 75, in _read_dates
    for node in sorted(deps_graph.nodes):
  File "/Users/jgsogo/dev/conan/conan/conans/client/graph/graph.py", line 113, in __lt__
    return self.__cmp__(other) == -1
  File "/Users/jgsogo/dev/conan/conan/conans/client/graph/graph.py", line 104, in __cmp__
    if self.conan_ref < other.conan_ref:
TypeError: '<' not supported between instances of 'str' and 'NoneType'
```

Maybe we need to ping @lasote, I'm not sure if I should return some kind of special revision here, or maybe the failing tests are related to the revisions feature (I haven't looked into it).

Reply:
Yes, seems an underlying issue with Py3 (py2 compares None with strings without problem), and the comparisons of references with and without revisions. It seems that such underlying issue should be investigated.

Reply:
@memsharded, please, tell us something about this `None/None@None/None`.